### PR TITLE
feat: add App Builder Template management commands

### PR DESCRIPTION
**Pull request by shazron**
## Description

Add app template commands. See `aio app template --help`
Support for non-npm registries are under the `--experimental-registry` flag under the `discover` command.

The experimental registry currently is at https://github.com/adobe/aio-app-builder-template-registry and you can use this [url](https://raw.githubusercontent.com/adobe/aio-app-builder-template-registry/main/data/metadata.json) for the `--experimental-registry` value.

## How Has This Been Tested?

Local link the plugin, and run the commands.

```bash
git clone https://github.com/adobe/aio-cli-plugin-app.git ACNA-1421
cd ACNA-1421
git checkout -b ACNA-1421 origin/ACNA-1421
npm i
aio plugins link .
# go to your App Builder app root, see available commands
aio app template --help
```

to unlink:
```bash
# go to your cloned repo root
cd ACNA-1421
aio plugins unlink .
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "eslint-plugin-promise": "^5.2.0",
     "eslint-plugin-standard": "^4.0.1",
     "jest": "^26.6.3",
+    "jest-fetch-mock": "^3.0.3",
```
Comment (shazron) [inline package.json:62]: note to self: can be removed

File: src/lib/npm-helper.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  const pjson = JSON.parse(fs.readFileSync(pjsonPath))
+
+  return pjson.version
+}
+
+/**
+ * Hide NPM Warnings by intercepting process.stderr.write stream
+ *
+ */
+function hideNPMWarnings () {
```
Comment (purplecabbage) [inline src/lib/npm-helper.js:173]: This seems like a bad practice, and has side-effects.
Can't we just use --loglevel=error for the `npm install`

**Review by purplecabbage — COMMENTED** [review]
This all looks good, however I think I would greatly prefer if we could split all `app template` commands into a separate `app-template` command, which still handled app:template commands identically, but this would allow us to pre-release / test this command independently.

File: src/lib/npm-helper.js (hunk 2 above)
Comment (shazron) [reply src/lib/npm-helper.js:173]: this was a copy paste from the discover plugin - to suppress warnings one level deeper that we have no control over (the plugin install command). Yes we should definitely change to what you recommended

**Comment by shazron** [comment]
> This all looks good, however I think I would greatly prefer if we could split all `app template` commands into a separate `app-template` command, which still handled app:template commands identically, but this would allow us to pre-release / test this command independently.

yup this is the plan going forward now

**Comment by shazron** [comment]
Closed with https://github.com/adobe/aio-cli-plugin-app-templates/pull/1

**Pull request closed without merging**